### PR TITLE
CORE-6935 Ledger module structure

### PR DESCRIPTION
**Pull request by lankydan**
Screenshots of the new modules, packages and classes can be seen below.

Screenshot 2022-10-17 at 12 48 37

Screenshot 2022-10-17 at 12 51 21

Screenshot 2022-10-17 at 12 51 52

Screenshot 2022-10-17 at 12 52 21

Screenshot 2022-10-17 at 12 52 50

### CODE REVIEW

File: components/ledger/ledger-consensual-flow/build.gradle [hunk 1]
```diff
@@ -1,38 +1,43 @@
+//buildscript {
+//    configurations.integrationTestRuntimeClasspath {
+//        exclude project(':testing:ledger:ledger-consensual-testkit')
+//    }
+//}
```
Comment (omfR3) [inline components/ledger/ledger-consensual-flow/build.gradle]: Is this a temporary thing while you're fixing the tests?

Comment (lankydan) [reply components/ledger/ledger-consensual-flow/build.gradle]: No, thanks for seeing that.

**Review by blsemo — DISMISSED** [review]
LGTM

**Comment by lankydan** [comment]
`SignableData` is now in `ledger-common-data`.

File: components/ledger/ledger-consensual-flow/src/test/kotlin/net/corda/ledger/consensual/flow/impl/transaction/serializer/kryo/ConsensualSignedTransactionKryoSerializerTest.kt [hunk 2]
```diff
@@ -21,7 +20,7 @@ import org.junit.jupiter.api.Assertions.assertEquals
 import org.junit.jupiter.api.Test
 import org.mockito.kotlin.mock
 
-class ConsensualSignedTransactionImplKryoSerializerTest {
+class  b {
```
Comment (vlajos) [inline components/ledger/ledger-consensual-flow/src/test/kotlin/net/corda/ledger/consensual/flow/impl/transaction/serializer/kryo/ConsensualSignedTransactionKryoSerializerTest.kt]: b?

Comment (lankydan) [reply components/ledger/ledger-consensual-flow/src/test/kotlin/net/corda/ledger/consensual/flow/impl/transaction/serializer/kryo/ConsensualSignedTransactionKryoSerializerTest.kt]: Yeah I just saw this, somehow typo'd that 😮‍💨

**Review by ronanbrowne — APPROVED** [review]
LGTM (build)

**Review by blsemo — APPROVED** [review]
LGTM

**Pull request merged**